### PR TITLE
Prepare Asset Swapper for RFQM

### PR DESCRIPTION
**Pull request by phil-ociraptor**
## Description

Contains a small amount of refactoring to make our existing code support both RFQT and RFQM flows. Unit tests have been updated and new coverage is added for RFQM flows.

This is MKR-3

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Comment by phil-ociraptor** [comment]
This is the final version of the draft PR https://github.com/0xProject/protocol/pull/175. This is simply a cleaned up version of that PR, with working unit tests.

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 1]
```diff
@@ -236,42 +238,15 @@ export class QuoteRequestor {
...
             takerToken,
             assetFillAmount,
             marketOperation,
             comparisonPrice,
             _opts,
-            'indicative',
+            this._rfqtAssetOfferings,
         );
-
-        // validate
```
Comment (alexkroeger) [inline packages/asset-swapper/src/utils/quote_requestor.ts:249]: I like the separation of concerns here

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 2]
```diff
@@ -509,4 +495,130 @@ export class QuoteRequestor {
...
+            assetFillAmount,
+            marketOperation,
+            comparisonPrice,
+            options,
+            'firm',
+            assetOfferings,
+        );
+        const quotes = quotesRaw.map(result => ({ ...result, response: result.response.signedOrder }));
+
+        // validate
```
Comment (alexkroeger) [inline packages/asset-swapper/src/utils/quote_requestor.ts:519]: nit: I would go even further and have a non-async validate quotes function, but not necessary

**Comment by phil-ociraptor** [comment]
(Testing Linear integration) This resolves MKR-3

**Review by alexkroeger — APPROVED** [review]

**Comment by phil-ociraptor** [comment]
Ok, I lied, this isn't the final PR. I'm going to close out this PR and refer to #187 as the canonical PR  as that one is linked properly w/Linear

**Pull request closed without merging**